### ISSUE TITLE
Memory leak with PerformanceCounter

### ISSUE THREAD

**Issue by ramdor** [ramdor/Thetis#15]
A process filtered PerformanceCounter leaks memory over time, and uses a large amount each time NextValue() is called. Tagged as MEMORYLEAK in source.

**Comment by dl2xy** [comment]
Cannot confirm. Thetis x64 running on WIN7 SP1 for several hours on "thetis only" without observable increase on mem usuage.

**Comment by ramdor** [comment]
> Cannot confirm. Thetis x64 running on WIN7 SP1 for several hours on "thetis only" without observable increase on mem usuage.

this only happens when using process filtered PerformaceCounters, such as the Thetis Only option in the status bar. It has been temporarily removed until a fix is found. It does however seem to be a .net/core issue, so may not get fix in the short term. System cpu% is fine, which uses the _Total performance counter.

Richie.

**Comment by dl2xy** [comment]
> this only happens when using process filtered PerformaceCounters, such as the Thetis Only option in the status bar. It has been temporarily removed until a fix is found. It does however seem to be a .net/core issue, so may not get fix in the short term. System cpu% is fine, which uses the _Total performance counter.

Yes, i have recompiled  with reverted changes and run with "Thetis only" option. (VS2019, target x64 release)

**Comment by ramdor** [comment]
> Yes, i have recompiled with reverted changes and run with "Thetis only" option. (VS2019, target x64 release)
> 
> Walter.

I ran it for about 6 hours, and noticed a gradual increase. Without it there is no gradual increase. I also noticed that memory use increases quite fast and then GC's back down again, but I feel there is perhaps something left over. If you set the CPU Meter (ms) in setup-Display-General to 100ms you can see this happen quite fast. It is probably best left out for now until there is more investigation on what is actually happening.

Richie.

**Comment by ramdor** [comment]
hmm, I wonder if it is a windows 11 issue?

![image](https://user-images.githubusercontent.com/3124967/162018807-0c2d2239-62e3-4dde-91b4-e479e3145c03.png)

**Comment by dl2xy** [comment]
On WIN8.1 it looks exactly the same.
But it resets every time to the same value, no long term increase. Maybe the sawtooth is just a garbage collection effekt.

Just testing WIN7...

**Comment by ramdor** [comment]
It is over 100MB of GC'ed just for a processcounter which is somewhat nasty. I have looked at the .net source but didn't really spot anything obvious. It doesn't really hinder much atm as there are so many other apps to check cpu usage of the process.

**Comment by dl2xy** [comment]
The 8.1 test was on QRL with an older build of thetis.
At home, WIN7 and self compiled thetis there is no sawtooth, Private Bytes stable at 748.6MB !

**Comment by ramdor** [comment]
It has to be a later windows issue in that case Walter. I will spend some time investigating that at some point, but low down on priority atm. At least you can get it working your end if you want to use the feature. Cheers, Richie.

**Comment by ramdor** [comment]
re-enabled in latest releases. Think this is a non issue now. We can re-open if needed.
